snapshotter/snapshotter.py

```diff
@@ -378,6 +394,13 @@ def snapshot(source, dest, debug=False, min_snapshots=3, extra_args=None):
     """
     date = _datetime()
     user, host, snapshots_root = _parse_path(dest)
+    if max_snapshots > -1 and max_snapshots - min_snapshots < 0:
+        raise InconsistentArgumentsError(
+            "--max-snapshots must be greater than --min-snapshots")
+    if max_snapshots > -1 and max_snapshots + 1 - min_snapshots >= 0:
+        snapshots = _ls_snapshots(dest)
+        for i in range(len(snapshots) - max_snapshots + 1):
+            _rm(snapshots[i], user, host, directory=True, debug=debug)
```

Comment:
I merged what you have done in my forked master.
The while is nice but is not working because max dedault value is -1, so in the case with no snapshot: 0 > -2...
I was thinking: maybe putting `max_snapshots` default value at "infinity", i.e in our case:  sys.maxint. It will reduce some tests and is more logical.